Swap ubuntu for alpine for terraform image build

mcrmonkey opened the pull request:
Reduces need for installation of additional packages to perform basic
tasks during build time.

Same image build process as packer. alpine pinned to 3.9.

leeonlee commented:
Hi, there was a recent merge that has some overlapping changes (using build_args to specify terraform version): https://github.com/GoogleCloudPlatform/cloud-builders-community/pull/279

Please resolve conflicts and I'll take a look, thanks!

mcrmonkey commented:
Conflicts, I believe, are now corrected...

I've also adjusted the `build-arg` options so that they aren't just comping out as `--build-arg 0.12.8` and instead are `--build-args TERRAFORM_VERSION=0.12.8`

leeonlee commented on terraform/Dockerfile:
```diff
@@ -1,21 +1,17 @@
+
+FROM gcr.io/cloud-builders/gcloud
 
 ENV PATH=/builder/terraform/:$PATH
```
Now that we're copying `terraform` to `/usr/bin/terraform`, this line is unneeded. I have no strong opinions on `/builder/terraform` vs `/usr/bin/terraform`.

mcrmonkey replied:
I've removed the `ENV` line but looks like the `RUN` and `FROM` lines are highlighted here :thinking:

leeonlee replied:
Yeah, the editor grabbed an extra couple of lines 🤷‍♀️

leeonlee commented on terraform/cloudbuild.yaml:
```diff
@@ -4,16 +4,18 @@ substitutions:
   - build
   - --build-arg
-  - TERRAFORM_VERSION
+  - TERRAFORM_VERSION=${_TERRAFORM_VERSION}
```
> I've also adjusted the `build-arg` options so that they aren't just comping out as `--build-arg 0.12.8` and instead are `--build-args TERRAFORM_VERSION=0.12.8`

I'm under the impression that the build-arg was not incorrect but I have no strong opinion here either: 

"You may also use the --build-arg flag without a value, in which case the value from the local environment will be propagated into the Docker container being built" [link](https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg)

leeonlee commented:
Thanks for the contribution!
